containers/unit-tests/scenario-dist.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh -eux
+
+if [ "${NO_NPM:-}" = "1" ]; then
+    # We can't 'make dist' with NO_NPM, so unset it and finish the build
+    unset NO_NPM
+    test ! -d /tmp/source/node_modules # this shouldn't be here
+    if [ -d /source/node_modules ]; then
+        cp -r /source/node_modules /tmp/source
+    fi
+    tools/npm-install
+    test -d /tmp/source/node_modules # this must surely be here now
+    make
+fi
+
+make XZ_COMPRESS_FLAGS='-0' NO_DIST_CACHE=1 dist
```

Comment:
Why `-0`?  If we're not disting the cache, we're gonna save a lot of time on the compression anyway, and a smaller tarball might be nicer...

Reply:
I mostly kept it as it was there before, and it does save a few minutes. But for this stage of the PR it's indeed not that relevant, it will become important only when integration tests get chained after this. Also note that the distcheck scenarios use this code as well. But I can take it out for now.